src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/DbColumnWrapper.cs

```diff
@@ -199,8 +199,8 @@ public DbColumnWrapper(DbColumn column)
                 object assemblyQualifiedName = column.UdtAssemblyQualifiedName;
                 const string hierarchyId = "MICROSOFT.SQLSERVER.TYPES.SQLHIERARCHYID";
 
-                if (assemblyQualifiedName != null &&
-                    string.Equals(assemblyQualifiedName.ToString(), hierarchyId, StringComparison.OrdinalIgnoreCase))
+                if (assemblyQualifiedName != null
+                && assemblyQualifiedName.ToString().StartsWith(hierarchyId, StringComparison.OrdinalIgnoreCase))
```

Comment:
I think this code is from SSMS, I found the original version, and it is using string.compare only to the length of hierarchyId string. bascially the same to StartWith. BTW, it seems StartWith is only available after .net 3.5.

Reply:
I could have missed that. When you see so many `string.Compare == 0` you start to just ignore the contents and replace with `string.Equals` blindly